modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/NodAndShuffleProgress.scala

```diff
@@ -24,6 +26,54 @@ import seqexec.web.client.semanticui._
 
 import scala.math.max
 
+final case class NodAndShuffleProgressMessage(
+                                               obsId : Observation.Id,
+                                               stepId: StepId,
+                                               fileId: ImageFileId,
+                                               stopping: Boolean,
+                                               paused: Boolean,
+                                               nsStatus: NodAndShuffleStatus
+                                             ) extends ReactProps {
+  @inline def render: VdomElement = NodAndShuffleProgressMessage.component(this)
+
+  protected[steps] val connect =
+    SeqexecCircuit.connect(SeqexecCircuit.obsProgressReader(obsId, stepId))
+}
+
+object NodAndShuffleProgressMessage extends ProgressLabel {
+  type Props = NodAndShuffleProgressMessage
+
+  @Lenses
+  protected case class State(progressConnect: ReactConnectProxy[Option[ObservationProgress]])

```

Comment:
Why is the connect in `State` Wouldn't be enough for it to live in the `Props` and make the component `stateless`?

Reply:
I added a comment about it in another use in the same file: https://github.com/gemini-hlsw/ocs3/blob/develop/modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/NodAndShuffleProgress.scala#L85

I followed the suggested way at the end of "Wrap and Connect Components" section in https://diode.suzaku.io/UsageWithReact.html.

If we keep in the props, when the props are recreated (which happens when a nod ends and another one starts), the component is unmounted and remounted.

Another option would be to have the `connect` it in a backend, but we would need to store it in a `var` and deal with initialization logic, since we need the `Props` to initialize it.